DevCycle/DevCycleClient.swift

```diff
@@ -406,23 +406,36 @@ public class DevCycleClient {
                         key: key,
                         value: nil,
                         defaultValue: defaultValue,
-                        evalReason: nil
+                        eval: EvalReason.defaultReason(details: DVCDefaultDetails.userNotTargeted.rawValue)
                     )
                 }
 
                 self.variableInstanceDictonary[key]?.setObject(
                     variable, forKey: defaultValue as AnyObject)
             }
 
-            if !self.closed && !self.disableAutomaticEventLogging {
+            if !self.closed && !self.disableAutomaticEventLogging {                
                 self.eventQueue.updateAggregateEvents(
-                    variableKey: variable.key, variableIsDefaulted: variable.isDefaulted)
+                    variableKey: variable.key,
+                    variableIsDefaulted: variable.isDefaulted,
+                    metadata: createVariableEventMetaData(variableEval: variable.eval)
+                )
             }
 
             return variable
         }
     }
 
+    private func createVariableEventMetaData(variableEval: EvalReason?) -> [String: [String: String]]? {
```

Comment:
Can we create a named type for this like `EvalMetaData`, so we can remove all the `metadata: [String: Any]?`

Reply:
added a `public typealias EvalMetaData = [String: Any]` and updated the usage everywhere